heron/metricsmgr/src/java/com/twitter/heron/metricsmgr/MetricsManager.java

```diff
@@ -328,24 +332,32 @@ public void start() {
      * Handler for uncaughtException
      */
     public void uncaughtException(Thread thread, Throwable exception) {
-      String threadName = thread.getName();
       LOG.log(Level.SEVERE,
-          "Exception caught in thread: " + threadName + " with thread id: " + thread.getId(),
+          "Exception caught in thread: " + thread.getName() + " with thread id: " + thread.getId(),
           exception);
 
-      String sinkId = threadName;
-      Integer thisSinkRetryAttempts = sinksRetryAttempts.remove(sinkId);
+      if (exception instanceof Error || thread.getId() == mainThreadId) {
+        LOG.severe("Would not recover from error. Metrics Manager halts immediately");
```

Comment:
My suggestion was instead of doing this:
```
- log all the details of the exception
if it's the main thread
  - log that we have to die (due to the previously logged exception)
```

to the log reader it will be more clear if we consolidate those two:
```
if it's the main thread
  - log all the details of the exception and that we have to die
else
  - log all the details of the exception
```